### PR TITLE
[MI-110] Add the Apps Resource

### PR DESCRIPTION
**Pull request by miogalang**
Add the Apps resource

/cc @jwswj @joseconsador @mmolina @atroche

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-110

### Risks
* Low. We might not be able to use the apps resources

### CODE REVIEW

**Comment by miogalang** [comment]
This is ready for review :haircut: @joseconsador  @jwswj @mmolina @atroche

**Comment by joseconsador** [comment]
:+1:

File: src/Zendesk/API/Resources/Apps.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    /**
+     * Uploads an app - see http://developer.zendesk.com/documentation/rest_api/apps.html for workflow
+     *
+     * @param array $params
+     *
+     * @throws \Exception
+     *
+     * @return mixed
+     */
+    public function upload(array $params)
```
Comment (jwswj) [inline src/Zendesk/API/Resources/Apps.php]: Seem this function a few times. Time for an abstraction?

**Comment by miogalang** [comment]
@jwswj :haircut:

**Comment by miogalang** [comment]
@jwswj :haircut:

**Comment by jwswj** [comment]
:+1:

**Pull request merged**